Makefile.coverage_targets

```diff
@@ -0,0 +1,15 @@
+collect-coverage: check-system
```

Comment:
This does not appear to fit with how we paralellize jobs in our CI.

Should this target perhaps refer to the presence of something that can potentially be collected through artifact exchange between jobs in a CI system?

Reply:
As mentioned above, initially I did not realize that this approach won't fit into our distributed CI runs. Perhaps we should remove these target entirely. It will simplify Makefile generation and we'll implement collation of coverage data in the github actions.